sdk/core/Azure.Core.TestFramework/src/TestRecording.cs

```diff
@@ -325,17 +328,25 @@ public string GetVariable(string variableName, string defaultValue, Func<string,
                 case RecordedTestMode.Live:
                     return defaultValue;
                 case RecordedTestMode.Playback:
-                    if (Variables.Count == 0)
-                    {
-                        throw new TestRecordingMismatchException("The recording contains no variables.");
-                    }
+                    ValidateVariables();
                     Variables.TryGetValue(variableName, out string value);
                     return value;
                 default:
                     throw new ArgumentOutOfRangeException();
             }
         }
 
+        private void ValidateVariables()
+        {
+            if (Variables.Count == 0)
+            {
+                throw new TestRecordingMismatchException(
+                    "The record session does not exist or is missing the Variables section. If the test is " +
```

Comment:
Is it possible to tell which of the two cases you are in?  Or will re-recording the test fix both cases?

Reply:
Yeah - rerecording would fix both.